modules/api/src/main/java/org/eclipse/xpanse/api/controllers/ServiceMigrationApi.java

```diff
@@ -120,6 +121,11 @@ private void validateEula(MigrateRequest migrateRequest) {
             log.error("Service not accepted Eula.");
             throw new EulaNotAccepted("Service not accepted Eula.");
         }
+        if (!existingServiceTemplate.getOcl().getBilling().getBillingModes()
+                .contains(migrateRequest.getBillingMode())) {
+            log.error("The service template does not support this billing mode.");
```

Comment:
It is better to output the template id and billingMode in the error message and then throw the error message.

Reply:
Good advice.